File: authorization-admin-service-spring/src/main/java/eu/rudisch/authorization/admin/repository/RoleRepository.java
package eu.rudisch.authorization.admin.repository;

import java.util.List;
import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;

import eu.rudisch.authorization.admin.model.Role;

public interface RoleRepository extends JpaRepository<Role, Integer> {
	String ROLE_AUTH_ADMIN = "ROLE_oauth_admin";

	Optional<Role> findByName(String name);

	@Query("select r from Role r where r.name in :names")
	List<Role> findByNames(@Param("names") List<String> names);
}

File: authorization-admin-service-spring-kotlin/src/main/kotlin/eu/rudisch/authorization/admin/repository/RoleRepository.kt
package eu.rudisch.authorization.admin.repository

import eu.rudisch.authorization.admin.model.Role
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param
import java.util.Optional

interface RoleRepository : JpaRepository<Role, Int> {
	companion object {
		const val ROLE_AUTH_ADMIN = "ROLE_oauth_admin"
	}

	fun findByName(name: String): Optional<Role>

	@Query("select r from Role r where r.name in :names")
	fun findByNames(@Param("names") names: List<String>): List<Role>
}